### DISCUSSION TITLE
having problem with impersonating an account on the dApps

### DISCUSSION THREAD

**Discussion by irnb** [paradigmxyz/rivet#89] [Q&A] [answered]
Hello everyone,

I recently tried out the **rivet**, and I must say, it's impressive and significantly simplifies the work for developers.

Here's a link to a presentation where Tom (@tmm) demonstrates it:  [https://www.youtube.com/live/aFk0WymgAo8?feature=shared&t=8775](https://www.youtube.com/live/aFk0WymgAo8?feature=shared&t=8775)

In it, he uses a **"Browser Wallet"** but I haven't been able to find any information about it on the internet. After some adjustments, it appears that this wallet is essential for **impersonating an account** when using other dApps, such as Uniswap.

Does anyone have more information about this **"Browser Wallet"**?

**Comment by irnb** [comment]
@jxom

**Comment by irnb** [comment, marked as answer]
Hi everyone,

I wanted to share a solution I found for those who are trying to use their browser wallet with Uniswap while using Rivet.

**Steps to Follow:**
1. **Disable Other Wallet Extensions:** First, ensure that all other wallet extensions in your browser are disabled, leaving only Rivet active.
2. **Configure Anvil with Correct ChainID:** Run your Anvil with `chainID` set to `1`. This is crucial as `1` corresponds to the Ethereum mainnet. If you're aiming to test on a different network, make sure to adjust the `chainID` to match that specific network's chain ID.

After following these steps, you should be able to see the `Browser Wallet` option available in Uniswap.

**Personal Experience:**
I've tested this setup on both Uniswap and Compound DApps. It worked flawlessly. I was able to impersonate a random account, execute transactions, and observe the results as if I were operating from that address.

**Acknowledgment:**
A big shoutout to the Rivet team for their excellent work. Your efforts are making a significant difference in the community!

Hope this helps others facing similar issues!

**Comment by Vanek131** [comment]
goood

**Comment by vikaman2103** [comment]
It often happens when you have to disable other browser wallets that would not pop up without a request
